interface/repository/pkmn_repository.go

```diff
@@ -0,0 +1,217 @@
+package repository
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/jvillarreal-w/academy-go-q42021/common"
+	"github.com/jvillarreal-w/academy-go-q42021/domain/model"
+	"github.com/jvillarreal-w/academy-go-q42021/usecase/repository"
+)
+
+type pokemonRepository struct {
+}
+
+func NewPokemonRepository() repository.PokemonRepository {
+	return &pokemonRepository{}
+}
+
+func readInternalDataSource(fileName string) ([][]string, error) {
+	f, err := os.Open(fileName)
+
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("Successfully opened CSV file")
+	defer f.Close()
+
+	rows, err := csv.NewReader(f).ReadAll()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
I think you would also like to return here

Reply:
Addressed as well, thank you.